client/lib/wp/sync-handler/test/cache-index.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import rewire from 'rewire';
+import mockery from 'mockery';
+import ms from 'ms';
+
+let cacheIndex, cacheIndexModule, localData, generateKey, RECORDS_LIST_KEY;
+
+const clearLocal = () => localData = {};
+const setRecordsList = recordsList => localData[ 'records-list' ] = recordsList;
+
+const localforageMock = {
+	getLocalForage() {
+		return {
+			setItem( key, data ) {
+				return new Promise( resolve => {
+					localData[ key ] = data;
+					resolve();
+				} )
+			},
+			getItem( key ) {
+				return new Promise( resolve => {
+					resolve( localData[ key ] );
+				} );
+			},
+			removeItem( key ) {
+				return new Promise( resolve => {
+					delete localData[ key ];
+					resolve();
+				} );
+			},
+			keys() {
+				return new Promise( resolve => {
+					resolve( Object.keys( localData ) );
+				} )
+			}
+		}
+	}
+};
+
+describe( 'cache-index', () => {
+	before( () => {
+		mockery.enable( {
+			warnOnReplace: false,
+			warnOnUnregistered: false
+		} );
+		mockery.registerMock( 'lib/localforage', localforageMock );
+		cacheIndexModule = rewire( '../cache-index' );
+		( { generateKey } = require( '../' ) );
+		cacheIndex = cacheIndexModule.cacheIndex;
+		RECORDS_LIST_KEY = cacheIndexModule.__get__( 'RECORDS_LIST_KEY' );
+	} );
+
+	after( function() {
+		mockery.disable();
+	} );
+
+	beforeEach( clearLocal ); // also do inside nested blocks with >1 test
+
+	describe( '#getAll', () => {
+		beforeEach( clearLocal );
+
+		it( 'should return empty array for empty localforage', () => {
+			return cacheIndex.getAll()
+			.then( ( res ) => {
+				expect( res ).to.equal( undefined );
+			} );
+		} );
+		it( 'should return index from localforage and nothing else', () => {
+			localData = {
+				test: 1,
+				test2: 2,
+			};
+			localData[ RECORDS_LIST_KEY ] = 3;
+			return cacheIndex.getAll()
+			.then( ( res ) => {
```

Comment:
Maybe `recordsList` is a better name than `res`. Also, I think it'd be helpful to use realistic values for the records list. Do we ever set that to an integer rather than an array of objects?